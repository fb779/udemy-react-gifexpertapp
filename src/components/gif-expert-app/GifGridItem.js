import React from 'react';
// import PropTypes from 'prop-types';

import './gifgriditem.css';

// const GifGridItem = ({id, title, small, medium, large}) => {
const GifGridItem = ({title, url}) => {
  return (
    <section className="grid-item animate__animated animate__fadeIn">
      <div>
        <img src={url} alt={title} />
      </div>
      <p>{title}</p>
    </section>
  );
};

// GifGridItem.propTypes = {};

export default GifGridItem;
