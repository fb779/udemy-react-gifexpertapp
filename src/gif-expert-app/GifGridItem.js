import React from 'react';
import PropTypes from 'prop-types';

import './gifgriditem.css';

// const GifGridItem = ({id, title, small, medium, large}) => {
const GifGridItem = ({title, url}) => {
  return (
    <section className="grid-item animate__animated animate__fadeIn">
      <img src={url} alt={title} />
      <p>{title}</p>
    </section>
  );
};

GifGridItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
};

export default GifGridItem;
