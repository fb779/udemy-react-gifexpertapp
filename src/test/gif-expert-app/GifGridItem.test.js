import {shallow} from 'enzyme';
import GifGridItem from '../../gif-expert-app/GifGridItem';

describe('Pruebas en componente <GifGridItem />', () => {
  const title = `Samuray X`;
  const url = `https://localhost.com/api/v1/imagen.png`;
  let wrapper = shallow(<GifGridItem title={title} url={url} />);

  beforeEach(() => {
    wrapper = shallow(<GifGridItem title={title} url={url} />);
  });

  test('debe cargar el componenete de forma adecuada', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('debe de tener un parrafo con el titulo', () => {
    const p = wrapper.find('p').text().trim();
    expect(p).toBe(title);
  });

  test('debe de tener una imagen con el src y el alt', () => {
    const img = wrapper.find('img');

    expect(img.prop('src')).toBe(url);
    expect(img.prop('alt')).toBe(title);
  });

  test('El section contenedor debe tener la claseName animated__fadeIn', () => {
    const section = wrapper.find('section');

    expect(section.prop('className').includes('animate__fadeIn')).toBe(true);
  });
});
