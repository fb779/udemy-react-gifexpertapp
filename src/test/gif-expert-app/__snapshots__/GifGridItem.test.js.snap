// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en componente <GifGridItem /> debe cargar el componenete de forma adecuada 1`] = `
<section
  className="grid-item animate__animated animate__fadeIn"
>
  <img
    alt="Samuray X"
    src="https://localhost.com/api/v1/imagen.png"
  />
  <p>
    Samuray X
  </p>
</section>
`;
